modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/RowVersion.java

```diff
@@ -38,10 +41,24 @@ public class RowVersion extends StorableBase {
      */
     public static final long NULL_LINK = 0;
 
+    private static final int TIMESTAMP_STORE_SIZE_BYTES = 2 * Long.BYTES;
+    private static final int NEXT_LINK_STORE_SIZE_BYTES = PartitionlessLinks.PARTITIONLESS_LINK_SIZE_BYTES;
+    private static final int VALUE_SIZE_STORE_SIZE_BYTES = Integer.BYTES;
+
+    public static final int TIMESTAMP_OFFSET = 0;
+    public static final int NEXT_LINK_OFFSET = TIMESTAMP_STORE_SIZE_BYTES;
+    public static final int VALUE_SIZE_OFFSET = NEXT_LINK_OFFSET + NEXT_LINK_STORE_SIZE_BYTES;
+    public static final int VALUE_OFFSET = VALUE_SIZE_OFFSET + VALUE_SIZE_STORE_SIZE_BYTES;
+
+    private final int partitionId;
```

Comment:
Can you please explain, why we need two fields if partition id can be derived from the link value?

Reply:
Because, when we create a new `RowVersion`, there is no yet link for it. That's why we need to provide the partition ID explicitly.